File: IntelliJ_Projects/src/MajorityElement/MajorityElementTest.java
package MajorityElement;

import QuickSelect.QuickSelect;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class MajorityElementTest {
    @Test
    public void MajorityElement_Test1() {
        int[] A = {2, 2, 2, 2, 3, 4, 1, 5, 2};
        assertEquals(MajorityElement.majority(A), 2);
    }

    @Test
    public void MajorityElement_Test2() {
        int[] A = {2, 2, 3, 4, 1, 5};
        assertEquals(MajorityElement.majority(A), -1);
    }
}

File: Python_code/MajorityElement/MajorityElementTest.py
import unittest

from MajorityElement import MajorityElement

class MajorityElementTest(unittest.TestCase):
    def MajorityElement_Test1(self):
        A = [2, 2, 2, 2, 3, 1, 4, 5, 2]
        self.assertEqual(MajorityElement().majority(A), 2)

    def MajorityElement_Test2(self):
        A = [2, 2, 3, 4, 1, 5]
        self.assertEqual(MajorityElement().majority(A), -1)

if __name__ == '__main__':
    unittest.main()